### PR TITLE
Ticket #2157: Use creator email rather than submitters for status updates / domain request

### LINKED ISSUE #2157: Change email sending to use Creator email not submitter on domain requests
### Issue description

On domain requests we send emails to the submitter email but we now need to send emails to the creator. However, this should be wrapped in the feature flag for user profile

### Acceptance criteria

- [ ] change email sending to use Creator email instead of submitter
- [ ] update and add to unit tests
- [ ] be sure this feature is wrapped in the feature flag

### Additional context

_No response_

### Links to other issues

blocked by: #1807
blocked by #2047
blocked by #2156
blocked by TBD tickets

### PR DESCRIPTION
**Pull request by zandercymatics**
## Ticket

Resolves #2157

## Changes

- For status change emails, we now send an email to self.creator rather than self.submitter when the flag "profile feature" is on
- Added onto existing unit tests and added one to test the flag

## Context for reviewers
This PR will send an email to the domain request _creator_ rather than the _submitter_ only if the profile feature flag is enabled

## Setup
For all of the steps below, make sure both the creator and submitter have emails that you are in control of
1. Change the status of any domain request with the profile flag OFF. You should expect an email to the submitter email
2. Change the status of any domain request with the profile flag ON. You should expect an email to the creator email

#### Satisfied acceptance criteria and met development standards

- [ ] Met the acceptance criteria, or will meet them in a subsequent PR
- [ ] Created/modified automated tests
- [ ] Added at least 2 developers as PR reviewers (only 1 will need to approve)
- [ ] Messaged on Slack or in standup to notify the team that a PR is ready for review
- [ ] Changes to “how we do things” are documented in READMEs and or onboarding guide
- [ ] If any model was updated to modify/add/delete columns, makemigrations was ran and the associated migrations file has been commited.

#### Ensured code standards are met (Original Developer)

- [ ] All new functions and methods are commented using plain language
- [ ] Did dependency updates in Pipfile also get changed in requirements.txt?
- [ ] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values

#### Validated user-facing changes (if applicable)

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)
- [ ] Add at least 1 designer as PR reviewer

#### Reviewed, tested, and left feedback about the changes

- [ ] Pulled this branch locally and tested it
- [ ] Reviewed this code and left comments
- [ ] Checked that all code is adequately covered by tests
- [ ] Made it clear which comments need to be addressed before this work is merged
- [ ] If any model was updated to modify/add/delete columns, makemigrations was ran and the associated migrations file has been commited.

#### Ensured code standards are met (Code reviewer)

- [ ] All new functions and methods are commented using plain language
- [ ] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values
- [ ] (Rarely needed) Did dependency updates in Pipfile also get changed in requirements.txt?

#### Validated user-facing changes as a developer

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Meets all designs and user flows provided by design/product
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [ ] Tested with multiple browsers, the suggestion is to use ones that the developer didn't (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

**Note:** Multiple code reviewers can share the checklists above, a second reviewers should not make a duplicate checklist

#### Verified that the changes match the design intention

- [ ] Checked that the design translated visually
- [ ] Checked behavior
- [ ] Checked different states (empty, one, some, error)
- [ ] Checked for landmarks, page heading structure, and links
- [ ] Tried to break the intended flow

#### Validated user-facing changes as a designer

- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [ ] Tested with multiple browsers (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

### CODE REVIEW

**Review by CocoByte — COMMENTED** [review]
Overall, code looks good -- I don't see the changes reflected in za-sandbox.  We can coordinate tomorrow on where to test this.

File: src/registrar/models/domain_request.py [hunk 1]
```diff
@@ -668,34 +669,42 @@ def delete_and_clean_up_domain(self, called_from):
...
+
+        wrap_email: bool -> Wraps emails using `wrap_text_and_preserve_paragraphs` if any given
+        paragraph exceeds our desired max length (for prettier display).
         """
 
-        if self.submitter is None or self.submitter.email is None:
-            logger.warning(f"Cannot send {new_status} email, no submitter email address.")
+        recipient = self.creator if flag_is_active(None, "profile_feature") else self.submitter
+        if recipient is None or recipient.email is None:
+            logger.warning(f"Cannot send {new_status} email, no creator email address.")
```
Comment (abroddrick) [inline src/registrar/models/domain_request.py:687]: Please add the requested domain name if it exists or at least the pk for the request.

**Review by abroddrick — APPROVED** [review]
Reviewed and looks good, had a weird error when running locally where changing creator did this 
image
However, this worked fine on sandbox so may just be my env. Also doesn't seem to relate to your ticket at all. I did not check every single status change though, that would be too much

**Review by CocoByte — APPROVED** [review]
LGTM! (tested and verified on getgov-za)

**Pull request merged**